src/plugins/data_source/server/client/configure_client.ts

```diff
@@ -172,10 +172,12 @@ const getQueryClient = async (
           ? await getAWSCredential(dataSourceAttr, cryptography!)
           : (dataSourceAttr.auth.credentials as SigV4Content));
 
-      const awsClient = rootClient ? rootClient : getAWSClient(credential, clientOptions);
-      addClientToPool(cacheKey, type, awsClient);
+      if (!rootClient) {
+        rootClient = getAWSClient(credential, clientOptions);
```

Comment:
for root client, let's not provide any credentials

Reply:
I agree. This is how `getAWSClient` function looks like. I'm just using region from `credential` as it required parameter while creating client. Should we refactor to only pass `region: string` parameter instead of `credential`. I might need to fix UT.

https://github.com/opensearch-project/OpenSearch-Dashboards/blob/d273c57081dd3e4652d8d40e613b8f25c72bd25f/src/plugins/data_source/server/client/configure_client.ts#L204-L213